src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDataSource.java

```diff
@@ -663,6 +663,24 @@ public int getQueryTimeout() {
                 SQLServerDriverIntProperty.QUERY_TIMEOUT.getDefaultValue());
     }
 
+    public void setEnablePrepareOnFirstPreparedStatementCall(boolean enablePrepareOnFirstPreparedStatementCall) {
+        setBooleanProperty(connectionProps, SQLServerDriverBooleanProperty.ENABLE_PREPARE_ON_FIRST_PREPARED_STATEMENT.toString(), enablePrepareOnFirstPreparedStatementCall);
+    }
+
```

Comment:
Can do. I added it to all places where there were javadocs already, DataSource didn't have them so just followed the pattern.